add content for CSS display property (#2188)
Codecademy/docs: content/css/concepts/display/display.md

  - 'paths/full-stack-engineer-career-path'
---

The CSS `display` property is a fundamental attribute that controls how an HTML element is rendered on a webpage. It determines the type of box used for an element and influences its layout and positioning within the document.

Understanding the different values of the `display` property is crucial for building well-structured and responsive web layouts. By choosing the appropriate value, you can control the flow of elements, create grids, and adjust the visibility of specific elements.

> 

Reviewer: You need to remove the following four blank lines.

  - 'paths/full-stack-engineer-career-path'
---

> # CSS Display Property

Reviewer: ```suggestion
```
We do not use h1 headers in Docs, nor do we have a title header. The Title field in the metadata takes care of that.

> 

Reviewer: ```suggestion
```
Remove extra line break.

The CSS `display` property is a fundamental attribute that controls how an HTML element is rendered on a webpage. It determines the type of box used for an element and influences its layout and positioning within the document.

> Understanding the different values of the `display` property is crucial for building well-structured and responsive web layouts. By choosing the appropriate value, you can control the flow of elements, create grids, and adjust the visibility of specific elements.

Reviewer: ```suggestion
Understanding the different values of the `display` property is crucial for building well-structured and responsive web layouts. By choosing the appropriate value, the flow of elements can be controlled, grids can be created, and the visibility of specific elements can be adjusted.
```
We do not use first or second person pronouns in Docs.

> In this guide, we will explore the commonly used values of the `display` property, provide examples of their usage, and discuss various considerations and techniques to maximize its potential.

Reviewer: ```suggestion
```
Remove this line

> 

Reviewer: ```suggestion
```
remove line break

> 

Reviewer: ```suggestion
```
remove line break

> 

Reviewer: ```suggestion
```
remove line break

  display: inline;
}

/* Overriding with higher specificity */
.container .element {
  display: block;
}
```

## Responsive design considerations

The display property can be combined with media queries to create responsive designs. Here's an example:

```css
@media (max-width: 768px) {
  .element {
    display: none; /* Hide element on smaller screens */
  }
}

@media (min-width: 768px) {
  .element {
    display: block; /* Display element on larger screens */
  }
}
```

## `display: grid` and `display: flex` Comparison

CSS Grid and Flexbox are powerful layout tools. The choice between display: flex and display: grid depends on the layout requirements and the desired design approach. 
Flexbox is particularly useful for creating responsive and flexible layouts, especially when dealing with content in a linear fashion, such as navigation menus, image galleries, or vertically aligned elements. 
CSS Grid is well-suited for creating advanced and flexible layouts, such as magazine-style designs, multi-column layouts, or designs that require explicit control over the placement of elements in a grid-like structure. Here is an example illustrating the difference between the two:

> ```css

Reviewer 2: For the tabbing in this example use 2 spaces instead of using tabs. So all the lines in the example will shift to the left after this is done.

      .flex-container {
        display: flex;
        justify-content: space-around;
      }
      .flex-item {
        background-color: cyan;
        padding: 20px;
        margin: 2px;
      }
      .grid-container {
        display: grid;
        grid-template-columns: 1fr 1fr 1fr;
        grid-template-rows: auto;
        grid-gap: 10px;
      }
      .grid-item {
        background-color: orange;
        padding: 20px;
      }
      .item-1 {
        grid-column: 1 / span 2;
        grid-row: 1;
      }
      .item-2 {
        grid-column: 3;
        grid-row: 1 / span 2;
      }
      .item-3 {
        grid-column: 1;
        grid-row: 2 / span 2;
      }
      .item-4 {
        grid-column: 2 / span 2;
        grid-row: 3;
      }
      .item-6 {
        grid-column: 1 / span 3;
      }
```

![Display Flex Versus Display Grid Example](https://raw.githubusercontent.com/Codecademy/docs/main/media/css-display-flex-versus-grid.png)

> ## Example 1

Reviewer 2: ```suggestion
## Example 
```

- `initial`
- `inherit`

## Differences Between Display Values

The display property in CSS allows you to control how elements are rendered on a webpage. Here are the commonly used display values and their descriptions:

### 1. `display: block`

Elements with `display: block;` are rendered as block-level elements. They create a line break after the element and occupy the full width of their parent container.

```css
.block-element {
  display: block;
}
```

### 2. `display: inline`

Elements with `display: inline;` are rendered as inline elements. They don't create line breaks and occupy only the necessary width and height based on their content.

```css
.inline-element {
  display: inline;
}
```

### 3.`display: inline-block`

Elements with `display: inline-block;` are rendered as inline-level block elements. They behave like inline elements but allow you to set width, height, padding, and margins.

```css
.inline-block-element {
  display: inline-block;
}
```

> ### 4. Display: none

Reviewer 2: ```suggestion
### 4. `display: none`
```

Elements with `display: none;` are not rendered and are completely hidden from the page. They don't take up any space in the document flow.

```css
.hidden-element {
  display: none;
}
```

> ### 5. Display: flex

Reviewer 2: ```suggestion
### 5. `display: flex`
```

- `initial`
- `inherit`

> ## Differences between display values

Reviewer: ```suggestion
## Differences between display values

```
Blank line after header

The display property in CSS allows you to control how elements are rendered on a webpage. Here are the commonly used display values and their descriptions:

### 1. Display: block

>  Elements with display: block; are rendered as block-level elements. They create a line break after the element and occupy the full width of their parent container.

Reviewer: ```suggestion
Elements with `display: block;` are rendered as block-level elements. They create a line break after the element and occupy the full width of their parent container.

```
Blank line between sentence and code block.

```css
.block-element {
  display: block;
}
```

> ### 2. Display: inline

Reviewer: ```suggestion
### 2. Display: inline

```
Blank line after header

> Elements with display: inline; are rendered as inline elements. They don't create line breaks and occupy only the necessary width and height based on their content.

Reviewer: ```suggestion
Elements with `display: inline;` are rendered as inline elements. They don't create line breaks and occupy only the necessary width and height based on their content.

```
Blank line between sentence and code block.

```css
.inline-element {
  display: inline;
}
```

### 3. Display: inline-block

>   Elements with display: inline-block; are rendered as inline-level block elements. They behave like inline elements but allow you to set width, height, padding, and margins.

Reviewer: ```suggestion
Elements with `display: inline-block;` are rendered as inline-level block elements. They behave like inline elements but allow you to set width, height, padding, and margins.

```
blank line between sentence and code block

```css
.inline-block-element {
  display: inline-block;
}
```

> ### 4. Display: none

Reviewer: ```suggestion

### 4. Display: none
```
Blank line between header and code block.

>  Elements with display: none; are not rendered and are completely hidden from the page. They don't take up any space in the document flow.

Reviewer: ```suggestion
Elements with `display: none;` are not rendered and are completely hidden from the page. They don't take up any space in the document flow.

```
Blank line between sentence and code block.

```css
.hidden-element {
  display: none;
}
```

> ### 5. Display: flex

Reviewer: ```suggestion

### 5. Display: flex
```
Blank line between code block and header

>  Elements with display: flex; create a flex container, allowing you to build flexible and responsive layouts. Flex items inside the container can be positioned and resized using flexbox properties.

Reviewer: ```suggestion
Elements with `display: flex;` create a flex container, allowing you to build flexible and responsive layouts. Flex items inside the container can be positioned and resized using flexbox properties.

```
Blank line between sentence and code block.

```css
.flex-container {
  display: flex;
}
```

> ### 6. Display: grid

Reviewer: ```suggestion

### 6. Display: grid
```
Blank line between code block and header.

> Elements with display: grid; create a grid container that enables you to define rows and columns, creating a two-dimensional grid system. Child elements inside the container can be placed in specific cells and adjusted for size and alignment.

Reviewer: ```suggestion
Elements with `display: grid;` create a grid container that enables you to define rows and columns, creating a two-dimensional grid system. Child elements inside the container can be placed in specific cells and adjusted for size and alignment.
```
Blank line between sentence and code block.

```css
.grid-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 10px;
}
```

> 

Reviewer: Remove 22 blank lines. (Only one blank line between the code block and the following header.

> ## Overriding display values:

Reviewer: ```suggestion
## Overriding display values
```
No ending punctuation on headers

CSS specificity allows you to override default display values of elements. Here's an example of overriding display:

```css
/* Original CSS */
.element {
  display: inline;
}

/* Overriding with higher specificity */
.container .element {
  display: block;
}

```

> ## Responsive design considerations:

Reviewer: ```suggestion
## Responsive design considerations

```
Blank line after header.

The display property can be combined with media queries to create responsive designs. Here's an example:

```css
@media (max-width: 768px) {
  .element {
    display: none; /* Hide element on smaller screens */
  }
}

@media (min-width: 768px) {
  .element {
    display: block; /* Display element on larger screens */
  }
}

```

> ## display: grid and display: flex comparison:

Reviewer: ```suggestion
## `display: grid` and `display: flex` Comparison

```

CSS Grid and Flexbox are powerful layout tools. Here's an example comparing the two:

> ```css

Reviewer: Can you provide an image illustrating this?

Author: Yes, I want to add this image, together with a proper description and the code.

![Illustrating display flex versus display grid](https://github.com/Codecademy/docs/assets/113944962/d44dc6b4-590c-4bea-b6cc-5593b8de9809)

I checked the documentation to make sure I am using the correct format and still my image is not being displayed. The image is located in the media folder and I inserted it using 

_![Display Flex Versus Display Grid Example](https://raw.githubusercontent.com/Codecademy/docs/main/media/css-display-flex-versus-grid.png)_

Still is not being displayed. Can you please help me understand what am I doing wrong? Thank you!

Reviewer 2: @Alexandra-Haynes You've linked it in properly. It won't display right now because it's not merged. Once merged, the image link will work and it will be displayed on the page.

Author: That makes sense! Thank you for the fast response, I appreciate your help

.grid-container {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 10px;
}

.flex-container {
  display: flex;
  justify-content: space-between;
}

```

> 

Reviewer: Remove two blank lines

- `initial`
- `inherit`

## Differences between display values:
```css
.block-element {
  display: block;
}

.inline-element {
  display: inline;
}

.inline-block-element {
  display: inline-block;
}

.hidden-element {
  display: none;
}

.flex-container {
  display: flex;
}

```

## Common use cases:

```css
/* Block-level element for creating a layout with stacked elements */
.block-element {
  display: block;
}

/* Inline element for inline text elements */
.inline-element {
  display: inline;
}

/* Inline block for inline elements that need block-level properties */
.inline-block-element {
  display: inline-block;
}

/* Hide an element from the layout */
.hidden-element {
  display: none;
}

/* Flexible container for creating responsive and flexible layouts */
.flex-container {
  display: flex;
}

```

## Overriding display values:

> 

Reviewer: ```suggestion
```
You need to remove instances where you have multiple line breaks in a row. There should never be more than one line break in a row.

> CSS specificity allows you to override default display values of elements. Here's an example of overriding display:

Reviewer: ```suggestion
CSS specificity allows you to override default display values of elements. Here's an example of overriding display:

```
There needs to be a line break before and after code blocks.

```css
/* Original CSS */
.element {
  display: inline;
}

/* Overriding with higher specificity */
.container .element {
  display: block;
}

```

> 

Reviewer: ```suggestion
```
Remove extra line break.

## Responsive design considerations:
The display property can be combined with media queries to create responsive designs. Here's an example:

```css
@media (max-width: 768px) {
  .element {
    display: none; /* Hide element on smaller screens */
  }
}

@media (min-width: 768px) {
  .element {
    display: block; /* Display element on larger screens */
  }
}

```

## display: grid and display: flex comparison:
CSS Grid and Flexbox are powerful layout tools. Here's an example comparing the two:

> ```css

Reviewer: This could use an illustrative image.

.grid-container {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 10px;
}

.flex-container {
  display: flex;
  justify-content: space-between;
}

```

> ---

Reviewer: ```suggestion
```
remove this

  display: flex;
}
```

> ---

Reviewer: ```suggestion
```
I would remove everything below this line.

  display: flex;
}
```

> 

Reviewer: Remove two blank lines.
